BreadPlayer.Views.UWP/Core/SharedLogic.cs

```diff
@@ -47,7 +51,26 @@ public SharedLogic()
             InitializeCore.IsMobile = ApiInformation.IsApiContractPresent("Windows.Phone.PhoneContract", 1);
 
             InitializeCore.IsMobile = Window.Current?.Bounds.Width <= 600;
+
+            MediaDevice.DefaultAudioRenderDeviceChanged += OnDefaultAudioRenderDeviceChanged;
+        }
+
+        private async void OnDefaultAudioRenderDeviceChanged(object sender, DefaultAudioRenderDeviceChangedEventArgs args)
+        {
+            if (args.Role != AudioDeviceRole.Default || args.Id == _audioDeviceId)
+                return;
+
+            var oldDevice = await DeviceInformation.CreateFromIdAsync(_audioDeviceId);
+            var device = await DeviceInformation.CreateFromIdAsync(args.Id);
+
+            System.Diagnostics.Debug.WriteLine($"Switching audio render device from [{oldDevice.Name}] to [{device.Name}]");
```

Comment:
Nice idea to log this but instead of or alongside `Debug` logging, there should also be `BPLogging`.